src/scripts/dot-gov.test.js

```diff
@@ -0,0 +1,331 @@
+      "Domain Type": "Interstate",
+      Organization: "Interstate Commission for EMS Personnel Practice",
+      "Security Contact Email": "(blank)",
+      State: "PA",
+    },
+    {
+      Agency: "Non-Federal Agency",
+      City: "St Croix",
+      "Domain Name": "VIVOTE.GOV",
+      "Domain Type": "State",
+      Organization: "Election System of the Virgin Islands",
+      "Security Contact Email": "(blank)",
+      State: "VI",
+    },
+    {
+      Agency: "Non-Federal Agency",
+      City: "Ada",
+      "Domain Name": "CHICKASAW-NSN.GOV",
+      "Domain Type": "Tribal",
+      Organization: "the Chickasaw Nation",
+      "Security Contact Email": "(blank)",
+      State: "OK",
+    },
+  ];
+  const commonResponse = {
+    icon_emoji: ":dotgov:",
+    thread_ts: "thread id",
+    unfurl_links: false,
+    unfurl_media: false,
+    username: ".Gov",
+  };
+
+  beforeAll(() => {});
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  afterAll(() => {});
+
+  it("subscribes to domain requests in two ways", () => {
+    script(app);
+    expect(app.message).toHaveBeenCalledWith(/git\.gov/i, expect.any(Function));
+    expect(app.message).toHaveBeenCalledWith(
+      expect.any(Function),
+      /((?<re_type>(Gov(ernment)?)|(City)|(County)|(Executive(\sBranch)?)|(Judicial(\sBranch)?)|(Legislative(\sBranch)?)|(Fed(eral)?)|((Ind(ependent)?\s)?Intra(state)?)|(Inter(state)?)|(State)|(Trib(e|(al))))\s)?((get\s?)?\.gov)(\s(?<re_search>.+))?/i,
+      expect.any(Function)
+    );
+  });
+
+  describe("response to domain requests", () => {
+    let handler;
+    beforeEach(() => {
+      script(app);
+      // general purpose handler, listens to all conversations
+      handler = app.getHandler();
+    });
+
+    const message = { message: { thread_ts: "thread id" }, say: jest.fn() };
+
+    it("fetches domains from a cache", async () => {
+      cache.mockResolvedValue([]);
+      await handler(message);
+      expect(cache).toHaveBeenCalledWith(
+        "dotgov domains",
+        1440,
+        expect.any(Function)
+      );
+    });
+
+    describe("gets domains from github if the cache is expired or whatever", () => {
+      let fetch;
+      beforeEach(async () => {
+        cache.mockResolvedValue([]);
+        await handler(message);
+        fetch = cache.mock.calls[0][2];
+      });
+
+      it("if the API throws an error", async () => {
+        axios.get.mockRejectedValue("error");
+        const out = await fetch();
+
+        expect(out).toEqual([]);
+      });
+
+      it("if the API returns a malformed object", async () => {
+        axios.get.mockResolvedValue({ data: { data_is_missing: [] } });
+        const out = await fetch();
+
+        expect(out).toEqual([]);
+      });
+
+      it("if the API doesn't return any domains", async () => {
+        axios.get.mockResolvedValue({ data: { data: [] } });
+        const out = await fetch();
+
+        expect(out).toEqual([]);
+      });
```

Comment:
If you're a completionist, you could add a test where the resolved value is just `{}` and doesn't have a `data` property at all. That generally shouldn't happen, but I suppose there's a case where the GitHub endpoint would return a 400 or 500 or something and no data – though I think that would cause Axios to throw an error, so maybe not? 🤷🏻‍♂️